components/share-workspace.vue

```diff
@@ -0,0 +1,74 @@
+<script setup lang="ts">
+import { toast } from "vue-sonner";
+import type { Workspace } from "~/server/db/schema";
+
+const props = defineProps<{
+  workspace: Workspace;
+}>();
+
+const { user } = useUserSession();
+
+const { activeInvitationLink, isLoading, createInvitationLink } =
+  useWorkspaceInvitationLink(props.workspace.id);
+
+const isOwner = computed(() => props.workspace.ownerId === user.value?.id);
+
+const fullLink = computed(() => {
+  const url = new URL(window.location.href);
+
+  url.pathname = "/api/invitation/accept";
+  url.searchParams.set("token", activeInvitationLink.value?.id ?? "");
+
+  return url.toString();
+});
```

Comment:
We might need to protect this against calling `window.*` during SSR. `useRequest` might allow us to access the request URL in SSR.